app/client/src/ce/sagas/userSagas.tsx

```diff
@@ -471,7 +471,7 @@ export function* logoutSaga(action: ReduxAction<{ redirectURL: string }>) {
       const currentUser: User | undefined = yield select(getCurrentUser);
       yield put(logoutUserSuccess(!!currentUser?.emptyInstance));
       localStorage.clear();
-      yield put(flushErrorsAndRedirect(redirectURL || AUTH_LOGIN_URL));
+      yield put(flushErrorsAndRedirect(redirectURL || AUTH_LOGIN_URL, true));
```

Comment:
I always figured we will clear the history and not create a new tab

```suggestion
history.go(-history.length) // This will take it to the first page of the session
history.replace(redirectURL || AUTH_LOGIN_URL) // This will show the correct page
```

Reply:
I actually started with that. But it doesn't seem to help.
Even though it successfully goes back to the login url, the history still exists.

Also tried `history.go(-(history.length - 1))` to move out of appsmith, but the result is same. Since the tab is still open the history persists.

https://www.loom.com/share/06243c58335f4ccf90588b37f604ff45

Reply:
Hence I thought we'll close the tab instead.

Reply:
As an alternative,
Perhaps we can restrict going back from signin page?
Instead of looking to clear history.

Reply:
Using alternate approach to prevent back button instead of opening a new tab

Reply:
Is it possible to clear history from the browser programmatically? Also I didn't get what Id's are sensitive in the url that we have to clear. Is it page id, app id etc? Also other apps don't clear anything in history I think for e.g notion post signout it momentarily show the page name and id, either ways this can seen from the browser history.